### ISSUE TITLE
TRACKING: Unholy Characters on some not all parts

### ISSUE THREAD

**Issue by 7ranceaddic7** [net-lisias-ksp/TweakScale#134]
These are two (2) of four (4) parts exhibiting this behavior.  The other two are structuralDisk1.25 & structuralDisk2.5.

Only these two (0.625 & 3.75) are complaining.

`[LOG 10:18:32.307] [TweakScale] INFO: NULL ConfigNode for StructuralDisks/Parts/structuralDisk0.625/structuralDisk0.625 (unholy characters on the name?). Trying partConfig instead!
[LOG 10:18:32.307] [TweakScale] INFO: NULL ConfigNode for StructuralDisks/Parts/structuralDisk0.625/structuralDisk0.625 (unholy characters on the name?). Trying partConfig instead!
[LOG 10:18:32.308] [TweakScale] INFO: NULL ConfigNode for StructuralDisks/Parts/structuralDisk0.625/structuralDisk0.625 (unholy characters on the name?). Trying partConfig instead!
[LOG 10:18:32.308] [TweakScale] INFO: NULL ConfigNode for StructuralDisks/Parts/structuralDisk3.75/structuralDisk3.75 (unholy characters on the name?). Trying partConfig instead!
[LOG 10:18:32.308] [TweakScale] INFO: NULL ConfigNode for StructuralDisks/Parts/structuralDisk3.75/structuralDisk3.75 (unholy characters on the name?). Trying partConfig instead!
[LOG 10:18:32.309] [TweakScale] INFO: NULL ConfigNode for StructuralDisks/Parts/structuralDisk3.75/structuralDisk3.75 (unholy characters on the name?). Trying partConfig instead!`

I've changed the periods (.) to underscores (_) in my game.  Will check back next time I start up a game.

**Comment by Lisias** [comment]
These are not necessarily problems, just annoyances (they are INFO nowadays, not even Warnings).

I don't use the Unity's URL subsystem for looking into things, waste of time most of the time. So when TweakScale reaches a part with that "unholy chars" (i.e., characters that are mangled on an URL), it look into the partConfig for the info instead.

Looking on the partConfig is less than ideal because without a source file, it's hard to debug things (and some synthetic parts may have no file!!), so I shove an entry on the KSP.log and if that part somehow is flagged later, that INFO line may hint why (usually we forgot that kraken damned URL transformation while accessing a kraken damned file!)

**Comment by 7ranceaddic7** [comment]
Here's an interesting thing...
```
[LOG 15:57:56.088] [TweakScale] INFO: NULL ConfigNode for StructuralDisks/Parts/structuralDisk0.625/structuralDisk0.625 (unholy characters on the name?). Trying partConfig instead!
[LOG 15:57:56.089] [TweakScale] INFO: NULL ConfigNode for StructuralDisks/Parts/structuralDisk0.625/structuralDisk0.625 (unholy characters on the name?). Trying partConfig instead!
[LOG 15:57:56.089] [TweakScale] INFO: NULL ConfigNode for StructuralDisks/Parts/structuralDisk0.625/structuralDisk0.625 (unholy characters on the name?). Trying partConfig instead!
[LOG 15:57:56.089] [TweakScale] INFO: NULL ConfigNode for StructuralDisks/Parts/structuralDisk3.75/structuralDisk3.75 (unholy characters on the name?). Trying partConfig instead!
[LOG 15:57:56.090] [TweakScale] INFO: NULL ConfigNode for StructuralDisks/Parts/structuralDisk3.75/structuralDisk3.75 (unholy characters on the name?). Trying partConfig instead!
[LOG 15:57:56.090] [TweakScale] INFO: NULL ConfigNode for StructuralDisks/Parts/structuralDisk3.75/structuralDisk3.75 (unholy characters on the name?). Trying partConfig instead!
```
What's interesting, we see that a lot, you say?

There are also these that don't complain...
StructuralDisks/Parts/structuralDisk1.25
StructuralDisks/Parts/structuralDisk2.5

So, why not?

But, wait, there's more ...

[LOG 15:57:56.003] [TweakScale] INFO: NULL ConfigNode for Squad/Parts/Engine/liquidEngineLV-1_v2/liquidEngineLV-1R _v2/radialEngineMini_v2 (unholy characters on the name?). Trying partConfig instead!

I get the others for using dots (.) in names, but this one REALLY doesn't make any sense.
	name = radialEngineMini_v2
There is no apparent reason for the UC label.

Digging.  More later.

**Comment by 7ranceaddic7** [comment]
Got the Squad prob solved. 

There's an extraneous space between the 1 and the underscore(_).

**Comment by Lisias** [comment]
Yep, I call these ones "unholy chars" for this exact reason - Forum rules would not allow logs to be published if I would use something... *more appropriate* ... :) for them. 

About this parts:

> StructuralDisks/Parts/structuralDisk1.25
> StructuralDisks/Parts/structuralDisk2.5

They are using dots. Dots are not mangled to anything. Underscores, on the other hand, are mangled into dots. So `StructuralDisks/Parts/structuralDisk1_25` will be translated to `StructuralDisks/Parts/structuralDisk1.25` completely disregarding if the later already exists on the file system.

> There's an extraneous space in the filename, between the 1 and the underscore(_).

What makes things yet more *interesting*, because spaces are not mangled and, on Module Manager, the only way to address a part with a space on it is using a wildcard. :P
